### ISSUE TITLE
firebase-messaging should be updated if firebase-core is updated to 16

### ISSUE THREAD

**Issue by jkasten2** [OneSignal/OneSignal-Gradle-Plugin#53]
## Issue

`com.google.firebase:firebase-messaging:15.0.2` is not compatible with `com.google.firebase:firebase-core:16.0.1`.
```Gradle
implementation 'com.google.firebase:firebase-messaging:15.0.2'
implementation 'com.google.firebase:firebase-core:16.0.1'
```

It results in the following runtime crash if this is the case.
```Java
07-12 14:50:39.586 18110-18790/com.onesignal.example E/AndroidRuntime: FATAL EXCEPTION: Firebase-MyFcmListenerService
    Process: com.onesignal.example, PID: 18110
    java.lang.NoSuchMethodError: No static method zzc(Landroid/content/Context;)Lcom/google/firebase/iid/zzs; in class Lcom/google/firebase/iid/zzs; or its super classes (declaration of 'com.google.firebase.iid.zzs' appears in /data/app/com.onesignal.example-BlapkZP7-S2w_MDm-qyX5w==/base.apk)
        at com.google.firebase.messaging.FirebaseMessagingService.zzd(Unknown Source:117)
        at com.google.firebase.iid.zzg.run(Unknown Source:26)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
        at com.google.android.gms.common.util.concurrent.zza.run(Unknown Source:7)
        at java.lang.Thread.run(Thread.java:764)
```

## Solution
`firebase-messaging` should be automatically updated to `17.0.0` if `firebase-core` is `16.0.0` or newer. This plugin currently has a [UPDATE_PARENT_ON_DEPENDENCY_UPGRADE](https://github.com/OneSignal/OneSignal-Gradle-Plugin/blob/017e7b5245f4edf8e51fcd4e987c3b202677b71b/src/main/groovy/com/onesignal/androidsdk/GradleProjectPlugin.groovy#L113) definition for minimum group version when other groups are updated, however it doesn't do this on the module level. The logic to support module level minimum should be similar to this however it should be a new hash set.

## Workaround
Upgrade or add the following to your project to fix the runtime crash noted above
```Gradle
implementation 'com.google.firebase:firebase-messaging:17.0.0'
```

**Comment by guyman** [comment]
This issue also crashes OneSignal for Cordova, any idea how to fix there ?

**Comment by jkasten2** [comment]
@guyman This is now fixed in the 0.11.1 update. For Cordova you may need to removing and re-add the Android platform if simply rebuild your project doesn't to get this updated automatically.
